test/extensions/filters/udp/dns_filter/dns_filter_fuzz_test.cc

```diff
@@ -0,0 +1,68 @@
+#include "common/common/logger.h"
+
+#include "extensions/filters/udp/dns_filter/dns_filter.h"
+
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/environment.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace DnsFilter {
+namespace {
+
+const std::string generateQuery(FuzzedDataProvider* data_provider) {
+  size_t query_size = data_provider->ConsumeIntegralInRange<size_t>(0, 512);
+  return data_provider->ConsumeRandomLengthString(query_size);
+}
+
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+  FuzzedDataProvider data_provider(buf, len);
+  const bool recurse = data_provider.ConsumeIntegralInRange<bool>(0, 1);
+  const uint16_t retry_count = data_provider.ConsumeIntegralInRange<uint16_t>(0, 3);
```

Comment:
Is this a limit just to constrain the fuzzer to not going haywire? if it's a reasonable constraint that should be respected, maybe a comment/validation in the API?

Reply:
Yes, I will add some validation in the API for this.    I believe the retry code is going to be exercised very little since most of the error handling is done when connecting to the DNS server.   It is essentially controlling a loop that will retry a name lookup if an error is returned by the external resolution callback.